packages/react/src/components/navigation/Navigation.tsx

```diff
@@ -173,15 +173,15 @@ export const Navigation = ({
   const [mobileMenuOpen, setMobileMenuOpen] = useState(menuOpen);
 
   const [authenticated, setAuthenticated] = useState(false);
-  const navigationVariant = getNavigationVariantFromChild(children);
+  const [navigationVariant, setNavigationVariant] = useState(getNavigationVariantFromChild(children));
```

Comment:
Should we pass callback to useState to not call the func on every render?
`useState(() => getNavigationVariantFromChild(children));`